plugins/main/public/components/endpoints-summary/agent/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { updateCurrentAgentData } from '../../../redux/actions/appStateActions';
+import { endpointSummary } from '../../../utils/applications';
+
+export const AgentView = () => {
+  const $injector = getAngularModule().$injector;
+
+  const $router = $injector.get('$route');
+  const $commonData = $injector.get('commonData');
+  const $reportingService = $injector.get('reportingService');
+
+  const shareAgent = new ShareAgent();
+
+  const savedTimefilter = $commonData.getTimefilter();
+  if (savedTimefilter) {
+    getDataPlugin().query.timefilter.timefilter.setTime(savedTimefilter);
+    $commonData.removeTimefilter();
+  }
+
+  const { agent: agentId } = $router.current.params;
+
+  const [agent, setAgent] = useState<Agent>();
+  const [isLoadingAgent, setIsLoadingAgent] = useState(true);
+  const [tab, setTab] = useState<string>($commonData.checkTabLocation());
+
+  const getAgent = async () => {
+    try {
+      setIsLoadingAgent(true);
+
+      const id = $commonData.checkLocationAgentId(
+        agentId,
+        shareAgent.getAgent(),
+      );
+
+      if (!id) {
+        return;
+      }
+
+      const { affected_items } = await getAgentsService({
+        agents: [agentId],
+      });
+      if (!affected_items?.length) {
+        throw 'Not found';
+      }
+
+      const agent = affected_items[0];
+      setAgent(agent);
+      store.dispatch(updateCurrentAgentData(agent));
+    } catch (error) {
+      const options = {
+        context: `AgentView.getAgent`,
+        level: UI_LOGGER_LEVELS.ERROR,
+        severity: UI_ERROR_SEVERITIES.CRITICAL,
+        store: true,
+        error: {
+          error,
+          message: error.message || error,
+          title: `Error getting the agent: ${error.message || error}`,
+        },
+      };
+      getErrorOrchestrator().handleError(options);
+    } finally {
+      setIsLoadingAgent(false);
+    }
+  };
+
+  useEffect(() => {
+    getAgent();
+  }, [tab]);
+
+  const exportConfiguration = enabledComponents => {
+    $reportingService.startConfigReport(
+      agent,
+      'agentConfig',
+      enabledComponents,
+    );
+  };
+
+  const switchTab = (tab: string) => {
+    setTab(tab);
+    getCore().application.navigateToApp(endpointSummary.id, {
+      path: `#/agents?tab=${tab}&agent=${agent?.id}`,
+    });
+  };
+
+  if (isLoadingAgent) {
+    return (
+      <EuiPage paddingSize='m'>
+        <EuiPageBody>
+          <EuiProgress size='xs' color='primary' />
+        </EuiPageBody>
+      </EuiPage>
+    );
+  }
+
+  if (tab === 'welcome') {
+    return <AgentsWelcome switchTab={switchTab} setAgent={setAgent} />;
+  }
+
+  if (tab === 'syscollector' && agent) {
+    return <MainSyscollector agent={agent} />;
+  }
+
+  if (tab === 'stats' && agent) {
+    return <MainAgentStats agent={agent} />;
+  }
+
+  if (tab === 'configuration' && agent) {
+    return (
+      <WzManagementConfiguration
+        agent={agent}
+        exportConfiguration={exportConfiguration}
```

Comment:
**thought**: this could not be used anymore because it internally uses the reporting service of ReactJS.